### PR TITLE
[feat] tools-v2: add snapshot-clone-status

### PR DESCRIPTION
**Pull request by Xinlong-Chen**
### What problem does this PR solve?

Issue Number: #2346

Problem Summary:  support `bs status snapshot` command in [curve tool](https://github.com/opencurve/curve/tree/master/tools-v2)

### What is changed and how it works?

What's Changed:  
1. add snapshot command(`snapshot.go`) in `curve/tools-v2/pkg/cli/command/curvebs/status` (also modify some code style in `etcd.go` and `mds.go`).
2. add new command reference in `tools-v2/pkg/cli/command/curvebs/status/status.go`.
3. add snapshot clone config in config file(`tools-v2/pkg/config`) and modify the user config file.

How it Works: using command `curve bs status snapshot`

Side effects(Breaking backward compatibility? Performance regression?): just client
the result in tools-v1:
```
SnapshotCloneServer status:
version: 9.9.9+2c4861ca
current snapshot-clone-server: 192.168.20.149:5550
online snapshot-clone-server list: 192.168.20.149:5550, 192.168.20.149:5551, 192.168.20.149:5552
offline snapshot-clone-server list:
```
the result in tools-v2:
```
+---------------------+---------------------+----------------+----------+
|        ADDR         |      DUMMYADDR      |    VERSION     |  STATUS  |
+---------------------+---------------------+----------------+----------+
| 192.168.20.149:5551 | 192.168.20.149:8101 | 9.9.9+2c4861ca | follower |
+---------------------+---------------------+                +          +
| 192.168.20.149:5552 | 192.168.20.149:8102 |                |          |
+---------------------+---------------------+                +----------+
| 192.168.20.149:5550 | 192.168.20.149:8100 |                | leader   |
+---------------------+---------------------+----------------+----------+
```

### Check List

- [x] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by Xinlong-Chen** [comment]
test success now! please take a look, thx.

File: tools-v2/pkg/config/bs.go [hunk 1]
```diff
@@ -36,57 +36,61 @@ import (
...
-	VIPER_CURVEBS_BURST_LENGTH   = "curvebs.burstlength"
-	CURVEBS_DEFAULT_BURST_LENGTH = uint64(10)
+	CURVEBS_MDSADDR                      = "mdsaddr"
+	VIPER_CURVEBS_MDSADDR                = "curvebs.mdsAddr"
+	CURVEBS_MDSDUMMYADDR                 = "mdsdummyaddr"
+	VIPER_CURVEBS_MDSDUMMYADDR           = "curvebs.mdsDummyAddr"
+	CURVEBS_SNAPSHOTCLONEADDR            = "snapshotCloneAddr"
+	VIPER_CURVEBS_SNAPSHOTCLONEADDR      = "curvebs.snapshotCloneAddr"
+	CURVEBS_SNAPSHOTCLONEDUMMYADDR       = "snapshotCloneDummyAddr"
+	VIPER_CURVEBS_SNAPSHOTCLONEDUMMYADDR = "curvebs.snapshotCloneDummyAddr"
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/config/bs.go:46]:
```suggestion
	CURVEBS_SNAPSHOTADDR            = "snapshotAddr"
	VIPER_CURVEBS_SNAPSHOTADDR      = "curvebs.snapshotAddr"
	CURVEBS_SNAPSHOTDUMMYADDR       = "snapshotDummyAddr"
	VIPER_CURVEBS_SNAPSHOTDUMMYADDR = "curvebs.snapshotDummyAddr"
```

File: tools-v2/pkg/config/bs.go [hunk 2]
```diff
@@ -96,27 +100,29 @@ var (
...
-		CURVEBS_STRIPE_COUNT:   VIPER_CURVEBS_STRIPE_COUNT,
-		CURVEBS_LIMIT:          VIPER_CURVEBS_LIMIT,
-		CURVEBS_BURST:          VIPER_CURVEBS_BURST,
-		CURVEBS_BURST_LENGTH:   VIPER_CURVEBS_BURST_LENGTH,
-		CURVEBS_FORCE:          VIPER_CURVEBS_FORCE,
-		CURVEBS_TYPE:           VIPER_CURVEBS_TYPE,
-		CURVE_EXPIRED_TIME:     VIPER_CURVE_EXPIRED_TIME,
-		CURVEBS_RECYCLE_PREFIX: VIPER_RECYCLE_PREFIX,
+		CURVEBS_SNAPSHOTCLONEADDR:      VIPER_CURVEBS_SNAPSHOTCLONEADDR,
+		CURVEBS_SNAPSHOTCLONEDUMMYADDR: VIPER_CURVEBS_SNAPSHOTCLONEDUMMYADDR,
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/config/bs.go:104]: ditto

File: tools-v2/pkg/cli/command/curvebs/status/etcd/etcd.go [hunk 3]
```diff
@@ -68,12 +68,13 @@ func NewEtcdCommand() *cobra.Command {
 }
 
 func (eCmd *EtcdCommand) AddFlags() {
-	config.AddBsEtcdAddrFlag(eCmd.Cmd)
 	config.AddHttpTimeoutFlag(eCmd.Cmd)
+	config.AddBsEtcdAddrFlag(eCmd.Cmd)
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/cli/command/curvebs/status/etcd/etcd.go:72]: no need

File: tools-v2/pkg/cli/command/curvebs/status/mds/mds.go [hunk 4]
```diff
@@ -59,8 +59,8 @@ func NewMdsCommand() *cobra.Command {
 }
 
 func (mCmd *MdsCommand) AddFlags() {
-	config.AddBsMdsFlagOption(mCmd.Cmd)
 	config.AddHttpTimeoutFlag(mCmd.Cmd)
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/cli/command/curvebs/status/mds/mds.go:62]: no need modify?

File: tools-v2/pkg/cli/command/curvebs/status/etcd/etcd.go (hunk 3 above)
Comment (Xinlong-Chen) [reply tools-v2/pkg/cli/command/curvebs/status/etcd/etcd.go:72]: > no need

Does the order of add config impact code logic?
I think this order is more good-looking...

File: tools-v2/pkg/cli/command/curvebs/status/mds/mds.go (hunk 4 above)
Comment (Xinlong-Chen) [reply tools-v2/pkg/cli/command/curvebs/status/mds/mds.go:62]: > no need modify?

same question as forementioned, I think this order is more good-looking...

Comment (Xinlong-Chen) [reply tools-v2/pkg/cli/command/curvebs/status/mds/mds.go:62]: done

File: tools-v2/pkg/cli/command/curvebs/status/etcd/etcd.go (hunk 3 above)
Comment (Xinlong-Chen) [reply tools-v2/pkg/cli/command/curvebs/status/etcd/etcd.go:72]: done

File: tools-v2/pkg/config/bs.go (hunk 1 above)
Comment (Xinlong-Chen) [reply tools-v2/pkg/config/bs.go]: done

File: tools-v2/pkg/config/bs.go (hunk 2 above)
Comment (Xinlong-Chen) [reply tools-v2/pkg/config/bs.go]: done

**Comment by Xinlong-Chen** [comment]
cicheck

**Comment by Xinlong-Chen** [comment]
cicheck

File: tools-v2/pkg/config/bs.go [hunk 5]
```diff
@@ -36,57 +36,61 @@ import (
...
-	VIPER_CURVEBS_BURST          = "curvebs.burst"
-	CURVEBS_DEFAULT_BURST        = uint64(30000)
-	CURVEBS_BURST_LENGTH         = "burstlength"
-	VIPER_CURVEBS_BURST_LENGTH   = "curvebs.burstlength"
-	CURVEBS_DEFAULT_BURST_LENGTH = uint64(10)
+	CURVEBS_MDSADDR                 = "mdsaddr"
+	VIPER_CURVEBS_MDSADDR           = "curvebs.mdsAddr"
+	CURVEBS_MDSDUMMYADDR            = "mdsdummyaddr"
+	VIPER_CURVEBS_MDSDUMMYADDR      = "curvebs.mdsDummyAddr"
+	CURVEBS_SNAPSHOTADDR            = "snapshotAddr"
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/config/bs.go:43]:
```suggestion
	CURVEBS_SNAPSHOTADDR            = "snapshotaddr"
```

**Comment by Xinlong-Chen** [comment]
cicheck

**Comment by Xinlong-Chen** [comment]
cicheck

File: tools-v2/pkg/config/bs.go (hunk 5 above)
Comment (Xinlong-Chen) [reply tools-v2/pkg/config/bs.go]: done

**Comment by Xinlong-Chen** [comment]
cicheck

File: tools-v2/README.md [hunk 6]
```diff
@@ -1057,6 +1057,30 @@ Output:
...
 ```
 
+#### staus snapshot
+
+get the snapshot clone status of curvefs
+
+Usage:
+
+```bash
+curve bs status snapshot
```
Comment (Cyber-SiKu) [inline tools-v2/README.md:1067]: add 
```markdown
|snapshot-clone-status|curve bs status snapshot|
```
in #comparison-of-old-and-new-commands/#curve-bs

File: tools-v2/pkg/config/bs.go [hunk 7]
```diff
@@ -36,57 +36,61 @@ import (
...
-	CURVEBS_BURST_LENGTH         = "burstlength"
-	VIPER_CURVEBS_BURST_LENGTH   = "curvebs.burstlength"
-	CURVEBS_DEFAULT_BURST_LENGTH = uint64(10)
+	CURVEBS_MDSADDR                 = "mdsaddr"
+	VIPER_CURVEBS_MDSADDR           = "curvebs.mdsAddr"
+	CURVEBS_MDSDUMMYADDR            = "mdsdummyaddr"
+	VIPER_CURVEBS_MDSDUMMYADDR      = "curvebs.mdsDummyAddr"
+	CURVEBS_SNAPSHOTADDR            = "snapshotaddr"
+	VIPER_CURVEBS_SNAPSHOTADDR      = "curvebs.snapshotAddr"
+	CURVEBS_SNAPSHOTDUMMYADDR       = "snapshotdummyAddr"
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/config/bs.go:45]:
```suggestion
	CURVEBS_SNAPSHOTDUMMYADDR       = "snapshotdummyaddr"
```

Comment (Xinlong-Chen) [reply tools-v2/pkg/config/bs.go]: done

**Comment by Xinlong-Chen** [comment]
recheck

**Comment by Xinlong-Chen** [comment]
cicheck

**Comment by Xinlong-Chen** [comment]
@Cyber-SiKu here, please have a look again...

File: tools-v2/README.md [hunk 8]
```diff
@@ -1057,6 +1057,30 @@ Output:
 +-------------------+-------------------+-------------------+----------+
 ```
 
+#### staus snapshot
+
+get the snapshot clone status of curvefs
```
Comment (aspirer) [inline tools-v2/README.md:1062]: s/curvefs/curvebs/

File: tools-v2/pkg/config/curve.yaml [hunk 9]
```diff
@@ -20,6 +20,8 @@ curvebs:
   mdsAddr: 127.0.0.1:6700,127.0.0.1:6701,127.0.0.1:6702   # __CURVEADM_TEMPLATE__ ${cluster_mds_addr} __CURVEADM_TEMPLATE__
   mdsDummyAddr: 127.0.0.1:7700,127.0.0.1:7701,127.0.0.1:7702  # __CURVEADM_TEMPLATE__ ${cluster_mds_dummy_addr} __CURVEADM_TEMPLATE__
   etcdAddr: 127.0.0.1:23790,127.0.0.1:23791,127.0.0.1:23792  # __CURVEADM_TEMPLATE__ ${cluster_etcd_addr} __CURVEADM_TEMPLATE__
+  snapshotAddr: 192.168.20.149:5550,192.168.20.149:5551,192.168.20.149:5552
```
Comment (aspirer) [inline tools-v2/pkg/config/curve.yaml:23]: add suffix: `# __CURVEADM_TEMPLATE__ ${cluster_etcd_addr} __CURVEADM_TEMPLATE__`
it's used by curveadm.

Comment (aspirer) [reply tools-v2/pkg/config/curve.yaml:23]: it's better `s/192.168.20.149/127.0.0.1/`

File: tools-v2/pkg/cli/command/curvebs/status/snapshot/snapshot.go [hunk 10]
```diff
@@ -0,0 +1,220 @@
...
+}
+
+func (sCmd *SnapshotCommand) ResultPlainOutput() error {
+	return output.FinalCmdOutputPlain(&sCmd.FinalCurveCmd)
+}
+
+func NewStatusSnapshotCommand() *SnapshotCommand {
+	snapshotCmd := &SnapshotCommand{
+		FinalCurveCmd: basecmd.FinalCurveCmd{
+			Use:     "snapshot",
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/cli/command/curvebs/status/snapshot/snapshot.go:200]:
```suggestion
			Use:     "snapshotserver",
```

File: tools-v2/README.md (hunk 6 above)
Comment (aspirer) [inline tools-v2/README.md:1067]: @Cyber-SiKu we'd better use a more meaningful cmd, which can distinguish volume snapshot progress and snapshot servers' status, and raft snapshot.

File: tools-v2/pkg/cli/command/curvebs/status/snapshot/snapshot.go (hunk 10 above)
Comment (Xinlong-Chen) [reply tools-v2/pkg/cli/command/curvebs/status/snapshot/snapshot.go:200]: done

File: tools-v2/README.md (hunk 8 above)
Comment (Xinlong-Chen) [reply tools-v2/README.md:1062]: done

File: tools-v2/pkg/config/curve.yaml (hunk 9 above)
Comment (Xinlong-Chen) [reply tools-v2/pkg/config/curve.yaml:23]: done

**Comment by Xinlong-Chen** [comment]
cicheck

**Comment by Xinlong-Chen** [comment]
cicheck

**Comment by Xinlong-Chen** [comment]
cicheck

**Comment by Xinlong-Chen** [comment]
cicheck

**Comment by Xinlong-Chen** [comment]
cicheck

**Comment by Xinlong-Chen** [comment]
cicheck

**Comment by Xinlong-Chen** [comment]
cicheck

**Comment by Xinlong-Chen** [comment]
cicheck

**Review by Cyber-SiKu — APPROVED** [review]

**Review by wu-hanqing — APPROVED** [review]

**Pull request merged**